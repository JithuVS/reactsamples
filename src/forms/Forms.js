function Forms() {
  const handleSubmit = (e) => {
    e.preventDefault();
  };

  return (
    <>
      <h4>Forms</h4>
      <div className="section">
        <form style={{ width: "100%" }} onSubmit={handleSubmit}>
          <fieldset
            style={{
              width: "60%",
              margin: "auto",
              marginBottom: "2%",
              textAlign: "end",
            }}
          >
            <legend>Personal Details</legend>
            <input
              className="fields"
              type="text"
              id="firstname"
              placeholder="Firstname"
              name="firstname"
              required
            />

            <input
              className="fields"
              type="text"
              id="lastname"
              placeholder="Lastname"
              name="lastname"
              required
            />

            <input
              className="fields"
              type="email"
              id="email"
              placeholder="Email"
              name="email"
              required
            />

            <input
              className="fields"
              type="tel"
              id="tel"
              placeholder="Mobile"
              name="tel"
              minLength="10"
              required
              pattern="[0-9]+"
            />

            <input
              className="fields"
              type="address"
              id="address"
              placeholder="Address"
              name="address"
              required
            />

            <input type="submit" className="buttons" style={{ width: "20%" }} />
          </fieldset>
        </form>
      </div>
    </>
  );
}

export default Forms;
