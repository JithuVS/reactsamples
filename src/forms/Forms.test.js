import renderer from "react-test-renderer";
import Forms from "./Forms";

it("renders correctly when there are no items", () => {
  const tree = renderer.create(<Forms />).toJSON();
  expect(tree).toMatchSnapshot();
});
