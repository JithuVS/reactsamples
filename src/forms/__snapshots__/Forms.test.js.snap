// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
Array [
  <h4>
    Forms
  </h4>,
  <div
    className="section"
  >
    <form
      onSubmit={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <fieldset
        style={
          Object {
            "margin": "auto",
            "marginBottom": "2%",
            "textAlign": "end",
            "width": "60%",
          }
        }
      >
        <legend>
          Personal Details
        </legend>
        <input
          className="fields"
          id="firstname"
          name="firstname"
          placeholder="Firstname"
          required={true}
          type="text"
        />
        <input
          className="fields"
          id="lastname"
          name="lastname"
          placeholder="Lastname"
          required={true}
          type="text"
        />
        <input
          className="fields"
          id="email"
          name="email"
          placeholder="Email"
          required={true}
          type="email"
        />
        <input
          className="fields"
          id="tel"
          minLength="10"
          name="tel"
          pattern="[0-9]+"
          placeholder="Mobile"
          required={true}
          type="tel"
        />
        <input
          className="fields"
          id="address"
          name="address"
          placeholder="Address"
          required={true}
          type="address"
        />
        <input
          className="buttons"
          style={
            Object {
              "width": "20%",
            }
          }
          type="submit"
        />
      </fieldset>
    </form>
  </div>,
]
`;
